### PR TITLE
Refs #81, #84 - Add support for backoff and deprecated public events.

### PR DESCRIPTION
**Pull request by n1k0**
This patch adds support for publicly subscribable `backoff` and `deprecated` events:

```js
const kinto = new Kinto();

kinto.events.on("deprecated", event => console.log(event.message));

kinto.events.on("backoff", releaseTime => {
  const releaseDate = new Date(releaseTime).toLocaleString();
  alert(`Backed off; wait until ${releaseDate} to retry`);
});
```

- [x] Implementation
- [x] Documentation
- [x] Dist files

Early feedback welcome @Natim @leplatrem

### CODE REVIEW

**Comment by n1k0** [comment]
@Natim r=?

File: src/index.js [hunk 1]
```diff
@@ -21,12 +25,12 @@ export default class Kinto {
       throw new Error("missing collection name");
 
     const bucket = this._options.bucket || DEFAULT_BUCKET_NAME;
-    const api = new Api(this._options.remote || "http://0.0.0.0:8888/v1", {
+    const api = new Api(this._options.remote || "http://0.0.0.0:8888/v1", this.events, {
```
Comment (Natim) [inline src/index.js]: nit not related to the PR, but I think we should use localhost:8000 instead of 0.0.0.0:8000

File: test/api_test.js [hunk 2]
```diff
@@ -28,31 +30,36 @@ describe("Api", () => {
...
 
     it("should accept a headers option", () => {
-      expect(new Api(`http://test/${SPV}`, {headers: {Foo: "Bar"}}).optionHeaders)
+      expect(new Api(`http://test/${SPV}`, events, {headers: {Foo: "Bar"}}).optionHeaders)
         .eql({Foo: "Bar"});
     });
 
     it("should validate protocol version", () => {
-      expect(() =>new Api(`http://test/v999`))
+      expect(() =>new Api(`http://test/v999`, events))
```
Comment (Natim) [inline test/api_test.js]: Shouldn't it be possible to deactivate events if events is undefined in the constructor?

Comment (n1k0) [inline test/api_test.js]: Yeah, I'll update to handle passed the `events` instance as an option for all classes.

**Comment by Natim** [comment]
r+

**Pull request merged**